invisible_cities/cities/buffy.py

```diff
@@ -0,0 +1,126 @@
+"""
+-----------------------------------------------------------------------
+                                Buffy
+-----------------------------------------------------------------------
+
+ -- Sorts MC sensor info into buffers, slays vampires etc
+
+This city reads nexus Monte Carlo Sensor information and sorts the
+information into data-like buffers without the addition of
+electronics noise.
+
+Uses a configured buffer length, pretrigger and threshold for the
+positioning of trigger like signal. If more than one trigger is found
+separated from each other by more than a buffer width the nexus event
+can be split into multiple data-like triggers.
+"""
+
+import numpy  as np
+import tables as tb
+
+from .. detsim.buffer_functions import      buffer_calculator
+from .. detsim.sensor_utils     import   first_and_last_times
+from .. detsim.sensor_utils     import          get_n_sensors
+from .. detsim.sensor_utils     import           sensor_order
+from .. detsim.sensor_utils     import pmt_and_sipm_bin_width
+from .. detsim.sensor_utils     import          trigger_times
+from .. io    .rwf_io           import          buffer_writer
+from .. reco                    import          tbl_functions as tbl
+
+from .. dataflow   import            dataflow as fl
+
+from .  components import                city
+from .  components import             collect
+from .  components import        copy_mc_info
+from .  components import mcsensors_from_file
+from .  components import         print_every
+from .  components import       signal_finder
+from .  components import           wf_binner
+
+
+@city
+def buffy(files_in     , file_out   , compression      , event_range,
+          print_mod    , detector_db, run_number       , max_time   ,
+          buffer_length, pre_trigger, trigger_threshold):
+
+    npmt, nsipm       = get_n_sensors(detector_db, run_number)
+    pmt_wid, sipm_wid = pmt_and_sipm_bin_width(files_in[0])
```

Comment:
If I understand correctly, reading of bin widths from nexus configuration table is repeated here and in the buffy source (though in the source you dont use `pmt_and_sipm_bin_width` explicitly)? So the `pmt_wid` and coroutine `pmt_binwid` arguments are the same? I assume it does not change per event basis so ideally we should always use `pmt_wid`?
On the other hand this line is not safe in case `files_in` is an empty list (it will fail with `IndexError` ) or in case `filesin[0]` is an empty file. The first one is easy to fix but not sure what is the best protection against the second one? Though unless the file is corrupt (and we dont have protections in IC for that) even empty nexus files will contain the config table?